src/payment/strategies/googlepay/googlepay-adyenv3-initializer.ts

```diff
@@ -0,0 +1,111 @@
+import { round } from 'lodash';
+
+import { Checkout } from '../../../checkout';
+import PaymentMethod from '../../payment-method';
+
+import { BillingAddressFormat, GooglePaymentData, GooglePayInitializer, GooglePayPaymentDataRequestV2, TokenizePayload } from './googlepay';
+
+export default class GooglePayAdyenV3Initializer implements GooglePayInitializer {
+    initialize(
+        checkout: Checkout,
+        paymentMethod: PaymentMethod,
+        hasShippingAddress: boolean
+    ): Promise<GooglePayPaymentDataRequestV2> {
+
+        return Promise.resolve(this._getGooglePayPaymentDataRequest(
+            checkout,
+            paymentMethod,
+            hasShippingAddress
+        ));
+    }
+
+    teardown(): Promise<void> {
+        return Promise.resolve();
+    }
+
+    parseResponse(paymentData: GooglePaymentData): Promise<TokenizePayload> {
+        const {
+            paymentMethodData: {
+                type,
+                tokenizationData: { token },
+                info: {
+                    cardNetwork: cardType,
+                    cardDetails: lastFour,
+                },
+            },
+        } = paymentData;
+
+        return Promise.resolve({
+            type,
+            nonce: token,
+            details: {
+                cardType,
+                lastFour,
+            },
+        });
+    }
+
+    private _getGooglePayPaymentDataRequest(
+        checkout: Checkout,
+        paymentMethod: PaymentMethod,
+        hasShippingAddress: boolean
+    ): GooglePayPaymentDataRequestV2 {
+        const {
+            outstandingBalance,
+            cart: {
+                currency: { code: currencyCode },
+            },
+        } = checkout;
+
+        const {
+            initializationData: {
+                gatewayMerchantId,
+                storeCountry: countryCode,
+                googleMerchantName: merchantName,
+                googleMerchantId: merchantId,
+                platformToken: authJwt,
+            },
+            supportedCards,
+        } = paymentMethod;
+
+        return {
+            apiVersion: 2,
```

Comment:
Same as above is the apiVersion 2?

Reply:
It's not connected with adyenv3, it's google pay api version